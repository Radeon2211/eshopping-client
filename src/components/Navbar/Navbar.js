import React from 'react';
import { Link } from 'react-router-dom';
import * as SC from './Navbar.sc';
import logo from '../../images/logo.png';
import Heading from '../UI/Heading/Heading';
import SearchForm from './SearchForm/SearchForm';
import SignedOutLinks from './SignedOutLinks/SignedOutLinks';
import SignedInLinks from './SignedInLinks/SignedInLinks';

const Navbar = (props) => {
  const { userProfile } = props;

  const authNav = userProfile ? <SignedInLinks username={userProfile.username} /> : <SignedOutLinks />;

  return (
    <SC.Wrapper>
      <Link to="/" className="header-link">
        <header className="header">
          <img src={logo} alt="E-Shopping" className="logo" />
          <Heading variant="h1" className="heading">shopping</Heading>
        </header>
      </Link>
      <SearchForm />
      {authNav}
    </SC.Wrapper>
  );
};

export default Navbar;
