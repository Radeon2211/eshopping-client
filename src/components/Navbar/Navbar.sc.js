import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.header`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.light1};
  box-shadow: 0 1px 5px rgba(0, 0, 0, 0.3);
  display: flex;
  justify-content: center;
  left: 0;
  padding: ${({ theme }) => theme.spacings.level1} ${({ theme }) => theme.spacings.level4};
  position: fixed;
  top: 0;
  width: 100%;
  z-index: ${({ theme }) => theme.zIndexes.level4};

  & > *:not(:last-child) {
    margin-right: ${({ theme }) => theme.spacings.level4};
  }

  & .heading {
    margin-left: ${({ theme }) => theme.spacings.level1};
  }

  & .logo {
    width: 4.6rem;
  }

  .header {
    display: flex;
  }

  @media only screen and (max-width: 75em) {
    & .heading {
      display: none;
    }
  }

  @media only screen and (max-width: 56.25em) {
    padding: ${({ theme }) => theme.spacings.level1} ${({ theme }) => theme.spacings.level3};

    & > *:not(:last-child) {
      margin-right: ${({ theme }) => theme.spacings.level3};
    }
  }

  @media only screen and (max-width: 37.5em) {
    flex-wrap: wrap;
    justify-content: space-between;

    & > *:nth-child(2) {
      margin-right: 0;
      margin-top: ${({ theme }) => theme.spacings.level1};
      width: 100%;
    }
  }
`;
