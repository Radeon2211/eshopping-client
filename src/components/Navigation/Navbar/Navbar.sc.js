import styled from 'styled-components';

export const Wrapper = styled.div`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.light1};
  display: flex;
  position: fixed;
  left: 0;
  top: 0;
  padding: ${({ theme }) => theme.spacings.level1} ${({ theme }) => theme.spacings.level4};
  width: 100%;
`;