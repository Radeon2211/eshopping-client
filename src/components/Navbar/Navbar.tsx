import { Link } from 'react-router-dom';
import * as SC from './Navbar.sc';
import SearchForm from './SearchForm/SearchForm';
import LoggedOutLinks from './LoggedOutLinks/LoggedOutLinks';
import LoggedInLinks from './LoggedInLinks/LoggedInLinks';
import { defaultAppPath } from '../../shared/constants';
import Header from '../UI/Header';
import { Profile } from '../../shared/types/types';

interface NavbarProps {
  userProfile: Profile | null | undefined;
}

export default function Navbar({ userProfile }: NavbarProps) {
  const authNav = userProfile ? (
    <LoggedInLinks username={userProfile.username} status={userProfile.status} />
  ) : (
    <LoggedOutLinks />
  );

  return (
    <SC.Wrapper data-testid="Navbar">
      <Link to={defaultAppPath} data-testid="Navbar-header-link">
        <Header size="small" />
      </Link>
      <SearchForm />
      {authNav}
    </SC.Wrapper>
  );
}
