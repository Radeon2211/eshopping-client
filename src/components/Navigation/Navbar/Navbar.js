import React from 'react';
import * as SC from './Navbar.sc';
import logo from '../../../images/logo.png';

const Navbar = () => {
  return (
    <SC.Wrapper>
      <header className="header">
        <img src={logo} alt="E-Shopping" />
      </header>
    </SC.Wrapper>
  );
};

export default Navbar;
