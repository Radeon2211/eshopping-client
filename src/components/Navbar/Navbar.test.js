import React from 'react';
import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Navbar from './Navbar';
import {
  defaultUserProfile,
  renderAppPart,
  testRouterPushCall,
} from '../../shared/testUtility/testUtility';

const mockStore = configureMockStore([thunk]);
const defaultStore = mockStore({
  auth: { cart: [] },
});

const setUp = (userProfile, searchParam = '?p=1', pushFn = jest.fn()) => {
  return {
    ...renderAppPart(<Navbar userProfile={userProfile} />, {
      pathname: '/products',
      search: searchParam,
      push: pushFn,
      store: defaultStore,
    }),
  };
};

describe('<Navbar />', () => {
  it('should render <LoggedInLinks /> if user is logged in', () => {
    setUp(defaultUserProfile);
    expect(screen.getByTestId('LoggedInLinks')).toBeInTheDocument();
  });

  it('should render <LoggedOutLinks /> if user is logged out', () => {
    setUp(null);
    expect(screen.getByTestId('LoggedOutLinks')).toBeInTheDocument();
  });

  it('should call push after clicking at logo if current path is other than default', () => {
    const pushFn = jest.fn();
    setUp(null, '?p=2', pushFn);
    fireEvent.click(screen.getByTestId('Navbar-header-link'));
    testRouterPushCall(pushFn, 0, '/products', '?p=1');
  });
});
