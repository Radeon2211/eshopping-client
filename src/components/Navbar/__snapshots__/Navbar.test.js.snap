// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Should render <LoggedInLinks /> if user is logged in 1`] = `
<DocumentFragment>
  <header
    class="sc-bdfBwQ btxKQt"
  >
    <a
      data-testid="Navbar-header-link"
    >
      <header
        class="header"
      >
        <img
          alt="E-Shopping"
          class="logo"
          src="logo.png"
        />
        <h1
          class="sc-gsTCUz gJcZNp heading"
        >
          shopping
        </h1>
      </header>
    </a>
    <form
      class="sc-dlfnbm cjBYFJ"
    >
      <input
        autocomplete="off"
        class="name-input"
        maxlength="150"
        name="product"
        spellcheck="false"
        type="text"
        value=""
      />
      <button
        class="sc-hKgILt fbPnWM"
        color="blue"
        type="submit"
      >
        search
      </button>
    </form>
    <nav
      class="sc-crrsfI btxiNp"
    >
      <a
        class="sc-kEjbxe dWwZfv"
        data-testid="CartLink"
      >
        <div
          class="sc-fubCfw bzVjbi"
          color=""
        >
          <svg>
            cart.svg
          </svg>
        </div>
      </a>
      <div
        class="sc-dQppl ggLoGe"
        data-testid="LoggedInLinks-user-box"
        id="user"
      >
        <span
          class="sc-gKsewC kqhhoC"
          color=""
          display="inline-block"
          overflow="hidden"
          size="3"
          spacing="0px"
          transform=""
        >
          username
        </span>
        <div
          class="sc-fubCfw koNpdQ"
          color=""
        >
          <svg>
            arrow.svg
          </svg>
        </div>
        <div />
      </div>
    </nav>
  </header>
</DocumentFragment>
`;

exports[`<Navbar /> Should render <LoggedOutLinks /> if user is logged out 1`] = `
<DocumentFragment>
  <header
    class="sc-bdfBwQ btxKQt"
  >
    <a
      data-testid="Navbar-header-link"
    >
      <header
        class="header"
      >
        <img
          alt="E-Shopping"
          class="logo"
          src="logo.png"
        />
        <h1
          class="sc-gsTCUz gJcZNp heading"
        >
          shopping
        </h1>
      </header>
    </a>
    <form
      class="sc-dlfnbm cjBYFJ"
    >
      <input
        autocomplete="off"
        class="name-input"
        maxlength="150"
        name="product"
        spellcheck="false"
        type="text"
        value=""
      />
      <button
        class="sc-hKgILt fbPnWM"
        color="blue"
        type="submit"
      >
        search
      </button>
    </form>
    <div
      class="sc-pFZIQ iLRtPl"
      direction="row"
      spacing="3"
      width="auto"
      wrap="nowrap"
    >
      <button
        class="sc-hKgILt dNINNA"
        color="blue"
        type="button"
      >
        login
      </button>
      <button
        class="sc-hKgILt fbPnWM"
        color="blue"
        type="button"
      >
        signup
      </button>
    </div>
  </header>
</DocumentFragment>
`;
